### PR TITLE
[GuestConfiguration] Get-AzureRmVmGuestPolicyReport cmdlet: Return detailed and latest reports by default

### PR DESCRIPTION
**Pull request by viveklingaiah**
## Description
 - Return only the latest and detailed reports for all policies assigned to VM (optionally, by initiative id or name) by default with Get-AzureRmVmGuestPolicy cmdlet
 - Remove Detailed and Latest switches of Get-AzureRmVmGuestPolicyReport cmdlet
- I will work on the second cmdlet Get-AzureRmVmGuestPolicyHistory or Get-AzureRmVmGuestPolicyStatusHistory, that would provide status history, each status would have report link in it

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by vivlingaiah** [comment]
get-azurermvmguestpolicyreport_changed
(screenshot of sample outputs of all parameter sets)

File: src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/ChangeLog.md [hunk 1]
```diff
@@ -17,5 +17,9 @@
     * Overview of change #1
         - Additional information about change #1
 -->
-## Current Release
- - First release of Get-AzureRmGuestConfigurationPolicy report cmdlet (version: 0.9.0-preview)
+## Current release - version: 0.9.1-preview
```
Comment (markcowl) [inline src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/ChangeLog.md]: Just put this data under Current release, and do not update these headings - versions are added by the semantic versioning tool

File: src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/help/Get-AzureRmVMGuestPolicyReport.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿---
+---
```
Comment (markcowl) [inline src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/help/Get-AzureRmVMGuestPolicyReport.md]: What about the added cmdlet for returning just the latest policy report?

File: src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/ChangeLog.md (hunk 1 above)
Comment (vivlingaiah) [reply src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/ChangeLog.md]: done

File: src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/help/Get-AzureRmVMGuestPolicyReport.md (hunk 2 above)
Comment (vivlingaiah) [reply src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/help/Get-AzureRmVMGuestPolicyReport.md]: - This cmdlet (Get-AzureRmVmGuestPolicyReport) in PR returns just the latest policy report
- I need to make some changes to backend APIs to implement the other cmdlet (Get-AzureRmVmGuestPolicyStatusHistory). We do not want to fetch detailed report for historical statuses.

Comment (mgreenegit) [reply src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/help/Get-AzureRmVMGuestPolicyReport.md]: No issues with moving this one forward while working on the second cmdlet.  In the sample report I see we are focused on single-resource results.  Before we release we should also review a sample report where there are >10 ComplianceReasons items for a Policy.  LMK if you need a VM id.

**Comment by maddieclayton** [comment]
@vivlingaiah Ping

**Comment by vivlingaiah** [comment]
@maddieclayton I'll send out another iteration soon today

**Comment by vivlingaiah** [comment]
@maddieclayton @markcowl Is there a way to implement "paging" behavior in cmdlets provided the APIs support paging ?

**Comment by vivlingaiah** [comment]
[Get-AzureRmVMGuestPolicyStatusHistory_CmdletOutput.txt](https://github.com/Azure/azure-powershell/files/2587164/Get-AzureRmVMGuestPolicyStatusHistory_CmdletOutput.txt)

Output of Get-AzureRmVMGuestPolicyStatusHistory Cmdlet

**Comment by vivlingaiah** [comment]
@mgreenegit with your comments,
1 - fixed
2 - I think not a GA blocker. In interest of going public, I won't fix it here. Will be considered as part of a different PR
3, 4 - Reasons not available yet from service. I cannot comment on it's availability. Won't fix / can't fix

**Comment by vivlingaiah** [comment]
@markcowl @maddieclayton Could you advise ? Build is failing due to below reason

capture

**Comment by maddieclayton** [comment]
@vivlingaiah Looks like preview was broken when merged to your branch - I'll get a PR out to fix your branch.

File: src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/help/Get-AzureRmVMGuestPolicyReport.md (hunk 2 above)
Comment (vivlingaiah) [reply src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/help/Get-AzureRmVMGuestPolicyReport.md]: Added the second cmdlet Get-AzureRmVMGuestPolicyStatusHistory. sample output on email

**Comment by vivlingaiah** [comment]
build passing now. @markcowl  @maddieclayton , could you review further please ? We would like to take these cmdlets to GA in 11/26 train.

File: src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/AzureRM.GuestConfiguration.psd1 [hunk 3]
```diff
@@ -51,7 +51,7 @@ DotNetFrameworkVersion = '4.5.2'
 # ProcessorArchitecture = ''
 
 # Modules that must be imported into the global environment prior to importing this module
-RequiredModules = @(@{ModuleName = 'AzureRM.Profile'; ModuleVersion = '5.6.0'; })
```
Comment (maddieclayton) [inline src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/AzureRM.GuestConfiguration.psd1]: Please add this reference back in

File: src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/ChangeLog.md [hunk 4]
```diff
@@ -17,5 +17,9 @@
     * Overview of change #1
         - Additional information about change #1
 -->
-## Current Release
- - First release of Get-AzureRmGuestConfigurationPolicy report cmdlet (version: 0.9.0-preview)
+## Current release 
+ - Return only the latest and detailed reports for all policies assigned to VM (optionally, by initiative id or name) by default with Get-AzureRmVmGuestPolicyReport cmdlet
+ - Remove Detailed and Latest switches of Get-AzureRmVmGuestPolicyReport cmdlet
+
+## 0.9.0-preview : September 2018
```
Comment (maddieclayton) [inline src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/ChangeLog.md]: Please only add to the changelog, do not add additional headers.

File: src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/help/Get-AzureRmVMGuestPolicyStatusHistory.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+## SYNOPSIS
+Gets guest configuration policy compliance status history for an initiative of type "Guest Configuration" that is assigned to a VM.
+An initiative is a policy of definition type "Initiative".
+
+## SYNTAX
+
+### VmScope (Default)
+```
+Get-AzureRmVMGuestPolicyStatusHistory [-ResourceGroupName] <String> [-VMName] <String> [-ShowOnlyChanges]
```
Comment (maddieclayton) [inline src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/help/Get-AzureRmVMGuestPolicyStatusHistory.md]: Please follow instructions here to suppress signature issues: https://github.com/Azure/azure-powershell/blob/preview/documentation/Debugging-StaticAnalysis-Errors.md

File: src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/AzureRM.GuestConfiguration.psd1 (hunk 3 above)
Comment (vivlingaiah) [reply src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/AzureRM.GuestConfiguration.psd1]: done

File: src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/help/Get-AzureRmVMGuestPolicyStatusHistory.md (hunk 5 above)
Comment (vivlingaiah) [reply src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/help/Get-AzureRmVMGuestPolicyStatusHistory.md]: changed. could you check ? Is this what needed

File: src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/ChangeLog.md (hunk 4 above)
Comment (vivlingaiah) [reply src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/ChangeLog.md]: changed. could you check ? Is this what needed

File: src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/help/Get-AzureRmVMGuestPolicyStatusHistory.md (hunk 5 above)
Comment (vivlingaiah) [reply src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/help/Get-AzureRmVMGuestPolicyStatusHistory.md]: fixed

**Comment by vivlingaiah** [comment]
@maddieclayton fixed signature issue and others as advised. Still failing. Could you check ? I'm seeing some errors in msbuild.err

**Pull request merged**